catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java

```diff
@@ -322,7 +323,8 @@ protected String generateAlterTableSql(
             updateColumnPositionFieldDefinition(updateColumnPosition, lazyLoadCreateTable));
       } else if (change instanceof TableChange.DeleteColumn) {
         TableChange.DeleteColumn deleteColumn = (TableChange.DeleteColumn) change;
-        alterSql.add(deleteColumnFieldDefinition(deleteColumn));
+        lazyLoadCreateTable = getOrCreateTable(databaseName, tableName, lazyLoadCreateTable);
+        alterSql.add(deleteColumnFieldDefinition(deleteColumn, lazyLoadCreateTable));
```

Comment:
I think you can judge whether the return value of `deleteColumnFieldDefinition` is empty, if it's not empty, we can then put it into `alterSql`;

Line `358` would not work if we do multiple changes at the same time.